force-app/main/default/classes/EditDistanceCalculator.cls

```diff
@@ -0,0 +1,45 @@
+public with sharing class EditDistanceCalculator 
+{
+    @TestVisible
+    private static Integer calculateEditDistance( String str1, String str2, Integer m, Integer n ) //NOPMD -  Have to use long parameter here
+    {
+		Contract.requires( ( str1 != null ), 'calculateEditDistance called with null string 1' );
+
+        Contract.requires( ( str2 != null ), 'calculateEditDistance called with null string 2' );
+
+        Contract.requires( ( m != null ), 'calculateEditDistance called with null length of string 1' );
+
+        Contract.requires( ( n!= null ), 'calculateEditDistance called with null length of string 2' );
+
+        if ( m == 0 )  // if str1 is empty - we will have to insert all values to str2 i.e insert operations = str2.length()
+        {
+           return n; 
+        }
+
+        if ( n == 0 ) // if str2 is empty - we will have to delete all values from str2 i.e delete operations = str1.length()
+        {
+            return m;
+        }
+
+        if ( str1.charAt( m - 1 ) == str2.charAt( n - 1 ) ) //if characters match 
+        {
+           return calculateEditDistance( str1, str2, m - 1, n - 1 );  // no change to edit distance: edit distance = previous
+        }
+
+        return Math.min( Math.min( calculateEditDistance( str1, str2, m, n - 1 ), // Insert
+                                   calculateEditDistance( str1, str2, m - 1, n ) ), // Remove
+                                   calculateEditDistance( str1, str2, m - 1, n - 1 ) // Replace
+                                 ) + 1;
```

Comment:
this line is not self-explanatory and hence the reader cannot understand whats going on here on just one glance of the code.  Follow the rule 'Keep it Simple and Stupid' while coding so that its clear to the reader without having to ask how does it work

Reply:
Okay, I'll try to make it simpler